src/adapter-manager.js

```diff
@@ -430,24 +432,28 @@ class AdapterManager extends EventEmitter {
         // adapter (i.e. for testing)
         const pluginClient = new PluginClient(manifest.name,
                                               {verbose: false});
-        pluginClient.register().then(adapterManagerProxy => {
+        try {
+          const adapterManagerProxy = await pluginClient.register();
           console.log('Loading adapter', manifest.name, 'as plugin');
-          const adapterLoader = dynamicRequire(path.join(adapterPath));
+          const adapterLoader = dynamicRequire(adapterPath);
           adapterLoader(adapterManagerProxy, newSettings, errorCallback);
-        }).catch(err => {
+        } catch (err) {
           console.error('Failed to register adapter with gateway');
           console.error(err);
-        });
+          return Promise.reject();
```

Comment:
Using await here feels awkward to me. I think that you can do something like this instead:
```
let promise = plugin.register();
promise.then(adapterManagerProxy => {
   ....other stuff ...
}).catch(err => {
   ... handle error ...
});
return pomise;
```
But I haven't been doing JS very long, so I'd defer to @benfrancis or @hobinjk

Reply:
It was previously `.then(...).catch(...)`, but I feel like `await` is a lot cleaner, given that the enclosing function is already async.

Reply:
The behavior is slightly different in this case because it previously swallowed the error. Returning `Promise.reject` reports the error in a way that `pluginClient.register().catch` didn't.

While using await isn't a significant saving in this scenario, I definitely think that it's worth converting this code to have consistency with other places in this file where async/await does save complexity.